hphp/runtime/vm/jit/vasm-lower.cpp

```diff
@@ -33,6 +33,52 @@ namespace HPHP { namespace jit {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+namespace detail {
+
+jit::vector<int> computeDominatingConstraints(Vunit& unit) {
+  jit::vector<int> vregConstraints(unit.blocks.size(), INT_MAX);
+
+  // Build reverse postorder list of blocks
+  auto const rpo = sortBlocks(unit);
+
+  // Calculate each block's dominating vreg constraint. The lowerers will use
+  // this as the initial vreg constraint for each block, adjusting as they
+  // encounter vregrestrict/vregunrestrict during lowering.
+  vregConstraints[rpo[0]] = 0;
+  for (auto b : rpo) {
+    // Iterate through the instructions updating the constraint accordingly.
+    auto con = vregConstraints[b];
+    for (size_t i = 0; i < unit.blocks[b].code.size(); ++i) {
+      auto& inst = unit.blocks[b].code[i];
```

Comment:
Just do a range-based for:

```
for (auto const& inst : unit.blocks[b].code) {
```

The explicit index iterators you see elsewhere are because we need to pass the index to someone, or because we're manipulating the vector and can't use standard iterators.  This is just a non-mutating loop, though.